sky/execution.py

```diff
@@ -200,6 +181,25 @@ def _execute(
                 down=down,
                 dryrun=dryrun,
             )) as dag:
+        dag.resolve_and_validate_volumes()
+        if (not _is_launched_by_jobs_controller and
+                not _is_launched_by_sky_serve_controller):
+            # Only process pre-mount operations on API server.
+            dag.pre_mount_volumes()
+        for task in dag.tasks:
+            if task.storage_mounts is not None:
+                for storage in task.storage_mounts.values():
+                    # Ensure the storage is constructed.
+                    storage.construct()
+            for resource in task.resources:
+                # For backward compatibility, we need to override the autostop
+                # config at server-side for legacy clients.
+                # TODO(aylei): remove this after we bump the API version.
+                resource.override_autostop_config(
+                    down=down, idle_minutes=idle_minutes_to_autostop)
+                if resource.autostop_config is not None:
+                    down = resource.autostop_config.down
+                    idle_minutes_to_autostop = resource.autostop_config.idle_minutes
```

Comment:
I think this still be processed before admin policy to allow user take the final decision of idle_minutes in the policy